enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+* There is a possibility that a cluster admin could mistakenly remove the services ConfigMap for an older WMCO version,
+  before Node upgrade is completed. This would result in WMCD not knowing what services it has configured on the VM.
+  In order to prevent this, WMCD can set the Windows service description to "openshift-managed", so that it
+  can search for services and remove them. In order to facilitate upgrades from a version without WMCD, this string
+  should be added to all existing services in 4.10, one major version ahead of WMCD's target release.
+
+## Design Details
+
+### Test Plan
+
+Most functionality described within this enhancement is already tested as part of the extensive end-to-end
+tests in the WMCO repo. As the different commands of WMCD are developed, they can be integrated into WMCO, and
+tested through the existing tests. New tests will be added to test functionality that wasn't already present, such
+as validating the state of the services ConfigMap.
+
+### Graduation Criteria
+
+This enhancement is targeted for OpenShift 4.11/ WMCO version 6.0.0.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+As the functionality described in this enhancement will be integrated into WMCO, the normal release process
+of WMCO will be followed. That being a community operator release for early preview, followed by the official Red Hat
+operator release.
+
+#### Removing a deprecated feature
+
+N/A, as the changes made in the enhancement should be invisible to the user.
+
+### Upgrade / Downgrade Strategy
+
+When WMCO is upgraded, it will create the new Windows service ConfigMap, and then for each existing Windows Node:
+1) Stop the WMCD service
+2) Copy over all new and modified files in the payload, including WMCD.
+3) Update the desired version annotation on the Node, to the current WMCO version.
+4) Start the WMCD service.
+
+WMCD, triggered by a mismatch between desired node version and current node version labels will correct the state
+by removing the Services specified in the previous version's ConfigMap, and using the ConfigMap of the version it is
+upgrading to, will configure the VM normally. As part of this process, the Node version label is patched from the
+previous version to the desired version.
+
+Downgrades are not supported through OLM, and thus are not supported by WMCO.
+
+### Version Skew Strategy
+
+This enhancement has the potential to reduce issues related to version skew with regard to Nodes, as the services
+ConfigMap provides a way for WMCD to upgrade a Node cleanly from one version to another. Otherwise, this remains the
+same.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+This remains the same, on failure, the state of Windows nodes will not be managed properly. With improved
+event generation problems will be easier to debug.
+
+#### Support Procedures
+
+The support procedures for WMCO will remain the same.
```

Comment:
Won't you have daemon logs to gather in addition? IOW, does this enhancement require a Windows specific change to `must-gather`?